components/cloud-event-gateway-proxy/pkg/health/probes.go

```diff
@@ -0,0 +1,30 @@
+package health
+
+import (
+	"net/http"
+)
+
+const (
+	livenessURI  = "/healthz"
+	readinessURI = "/readyz"
+)
+
+func WithLivenessCheck(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		if request.RequestURI == livenessURI {
+			writer.WriteHeader(http.StatusOK)
+			return
+		}
+		next.ServeHTTP(writer, request)
+	})
+}
+
+func WithReadinessCheck(next http.Handler) http.Handler {
```

Comment:
In the Readiness probe check, shall we also add the connection status of proxy to the cloudevent gateway?

Reply:
At the moment that is not possible without actually trying to publish to the EMS gateway. Maybe in the future we will have some health check endpoint from EMS to ping.

Reply:
how about a ping?